pkg/pipelines/processor/remoteTasks_migrator.go

```diff
@@ -0,0 +1,484 @@
+		// and so we should migrate it to a new pipeline run
+		log.Logger().Infof("StepTemplate has image \"%s\". Migrating to new pipelineRun", stepTemplate.Image)
+		return p.MigrateToNewPipelineRun(prs, path)
+	}
+
+	// Otherwise we can assume that the pipeline run is a parent pipeline run, and so we should migrate it to individual tasks
+	log.Logger().Infof("StepTemplate has image \"%s\". Migrating to individual tasks", stepTemplate.Image)
+	return p.MigrateToTasks(prs, path)
+}
+
+type parentReference struct {
+	URL        string
+	Revision   string
+	PathInRepo string
+}
+
+func (pr parentReference) isEmpty() bool {
+	return pr.URL == "" && pr.Revision == "" && pr.PathInRepo == ""
+}
+
+// fileName returns the filename of the parent pipeline run
+func (pr parentReference) fileName() string {
+	return strings.TrimSuffix(filepath.Base(pr.PathInRepo), ".yaml")
+}
+
+// getParentReference parses the image name to get the parent pipeline run URL and revision
+func newParentReference(image, stepName string) parentReference {
+	if !strings.HasPrefix(image, "uses:") {
+		return parentReference{}
+	}
+	image = strings.TrimPrefix(image, "uses:")
+	split := strings.Split(image, "@")
+	fullURL, revision := split[0], split[1]
+
+	// Todo: REMOVE THIS HACK
+	if revision == "versionStream" {
+		revision = "master"
+	}
+
+	// Todo: And this hack
+	fullURL = strings.Replace(fullURL, "jenkins-x", "spring-financial-group", 1)
+
+	if !strings.HasPrefix(fullURL, "https://") {
+		// If the URL doesn't start with https:// then we can assume that it is a GitHub URL and so prepend that domain
+		fullURL = fmt.Sprintf("https://github.com/%s", fullURL)
+	}
+
+	split = strings.SplitAfter(fullURL, "/")
+	repoURL := strings.Join(split[:5], "")
+	repoURL = strings.TrimSuffix(repoURL, "/") + ".git"
+
+	pathInRepo := strings.Join(split[5:], "")
+	if stepName != "" {
+		pathInRepo = filepath.Join(strings.TrimSuffix(pathInRepo, ".yaml"), stepName) + ".yaml"
+	}
+
+	return parentReference{
+		URL:        repoURL,
+		Revision:   revision,
+		PathInRepo: pathInRepo,
+	}
+}
+
+func (p *RemoteTasksMigrator) MigrateToNewPipelineRun(prs *v1beta1.PipelineRun, path string) (bool, error) {
+	steps := prs.Spec.PipelineSpec.Tasks[0].TaskSpec.Steps
+	newPrs := v1beta1.PipelineRun{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "tekton.dev/v1beta1",
+			Kind:       "PipelineRun",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: prs.Name,
+		},
+		Spec: v1beta1.PipelineRunSpec{
+			PipelineSpec: &v1beta1.PipelineSpec{},
+			Workspaces: []v1beta1.WorkspaceBinding{
+				{
+					Name: "pipeline-ws",
+					VolumeClaimTemplate: &v1.PersistentVolumeClaim{
+						Spec: v1.PersistentVolumeClaimSpec{
+							AccessModes: []v1.PersistentVolumeAccessMode{
+								v1.ReadWriteOnce,
+							},
+							Resources: v1.ResourceRequirements{
+								Requests: v1.ResourceList{
+									v1.ResourceStorage: resource.MustParse("1Gi"),
```

Comment:
While 1Gi is a reasonable default it wouldn't be enough for all of our repos for example. This should be noted in documentation that it might need to be changed and more ambitiously the size of the repo could be checked and rounded upwards with a margin of a few hundred Mi (to accommodate for build artefacts).

Reply:
I've added the workspace size as a flag to the command so then if people are converting existing repos they can override the value (I'll add this info to the docs). For use in the catalog I think 1Gi is reasonable. I'll look in the future to add some `calculate workspace size` command that can be used for updating the workspace size as the repo grows.

Reply:
I've implemented your idea. Default functionality is to calculate the workspace size based on the size of the repository plus a 300Mi buffer for additional files.